com/course/zenkouv/src/main/java/lecture5/util/reader/Reader.java

```diff
@@ -0,0 +1,46 @@
+package lecture5.util.reader;
+
+import lecture5.util.validator.ExpressionValidator;
+
+import java.util.Scanner;
+
+public class Reader {
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    private Reader() {
+    }
+
+    public static double readNumber() {
+        String number;
+        while (true) {
+            System.out.print("Enter variable: ");
+            number = SCANNER.next();
+            if (ExpressionValidator.isNumber(number)) {
+                return Double.valueOf(number);
+            }
```

Comment:
print error message in case if user provide invalid value - to better user friendly API

Reply:
fix